src/Octokit.Webhooks/Converter/DateTimeStringConverter.cs

```diff
@@ -0,0 +1,40 @@
+namespace Octokit.Webhooks.Converter
+{
+    using System;
+    using System.Globalization;
+    using System.Text.Json;
+    using System.Text.Json.Serialization;
+
+    public class DateTimeStringConverter : JsonConverter<string>
+    {
+        private const string DateTimeStringFormat = "yyyy-MM-ddTHH:mm:ssK";
+        private static readonly DateTime UnixTimestampZero = new(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
+
+        public override string Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) => reader.TokenType switch
+        {
+            JsonTokenType.String => HandleString(reader),
+            JsonTokenType.Number => HandleNumber(reader),
+            _ => throw new JsonException($"'{reader.TokenType}' isn't an allowed token type for the {nameof(DateTimeStringConverter)}"),
+        };
+
+        public override void Write(Utf8JsonWriter writer, string value, JsonSerializerOptions options) => writer.WriteStringValue(value);
+
+        private static string HandleString(Utf8JsonReader reader)
+        {
+            var stringValue = reader.GetString();
+
+            if (stringValue == null)
+            {
+                throw new JsonException("Unable to convert string json token");
+            }
+
+            return stringValue;
+        }
+
+        private static string HandleNumber(Utf8JsonReader reader)
+        {
+            var dateTime = UnixTimestampZero.AddSeconds(reader.GetInt32());
+            return dateTime.ToString(DateTimeStringFormat, CultureInfo.InvariantCulture);
+        }
```

Comment:
What do you think about simplifying to something like this?

```suggestion
        private static string HandleNumber(Utf8JsonReader reader) =>
        	DateTimeOffset.FromUnixTimeSeconds(reader.GetInt32()).ToString(CultureInfo.InvariantCulture);
```

Reply:
Hey, the use of `DateTimeOffset.FromUnixTimeSeconds` seems to result in a `"2022-01-07T22:36:26+00:00"` format by default which is slightly different from all the other DateTime strings sent from the API (`"2022-01-07 22:36:26Z"`). To get them to be the same, it could become:
```cs
private static string HandleNumber(Utf8JsonReader reader) =>
            DateTimeOffset.FromUnixTimeSeconds(reader.GetInt32()).ToString("u", CultureInfo.InvariantCulture);
```
Note the `"u"` format in the `.ToString()`, though I'd rather the `"u"` lived in a class-level const with a name which describes why it's there/what it does.

Got any thoughts? :)

Reply:
See my comment below about deserializing to `DateTimeOffset`